compare/src/components/molecules/ApproveButton.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { approveTest, filterTests } from '../../actions';
+import { colors, fonts } from '../../styles';
+
+const REMOTE_HOST = 'http://127.0.0.1';
+const REMOTE_PORT = 3000;
+
+const Button = styled.button`
+  font-size: 20px;
+  line-height: 40px;
+  font-family: ${fonts.latoRegular};
+  background-color: ${colors.green};
+  border: none;
+  border-radius: 3px;
+  color: ${colors.white};
+  padding: 0px 30px;
+  margin-bottom: 10px;
+  
+  &:hover {
+    cursor: pointer;
+  }
+
+  &:disabled {
+    background-color: ${colors.bodyColor};
+    color: ${colors.secondaryText};
+    cursor: default;
+  }
+`;
+
+const ErrorMsg = styled.p`
+  word-wrap: break-word;
+  font-family: monospace;
+  background: rgb(251, 234, 234);
+  padding: 2ex;
+  color: brown;
+`;
+
+const APPROVE_STATUS_TO_LABEL_MAP = {
+  initial: 'Approve',
+  pending: 'Pending...',
+  approved: 'Approved',
+  failed: 'Approve'
+};
+
+class ApproveButton extends React.Component {
+  constructor (props) {
+    super(props);
+    this.approve = this.approve.bind(this);
+    this.state = {
+      approveStatus: 'initial',
+      errorMsg: null
+    };
+  }
+
+  approve () {
+    const { fileName } = this.props;
+    const url = `${REMOTE_HOST}:${REMOTE_PORT}/approve?filter=${fileName}`;
+    this.setState({ approveStatus: 'pending' });
+
+    fetch(url, {
+      method: 'POST'
+    })
+      .then(response => {
+        if (response.ok) {
+          this.setState({ approveStatus: 'approved' });
```

Comment:
Can this string literal be imported from a common mapping or constants location `approved`? instead of hard coding it in multiple places(there is at least one more occurence down this file) ?

Reply:
Hi Shankar, I am using a map 'APPROVE_STATUS_TO_LABEL_MAP'. 'approved' here is the key of the map.

Reply:
👍 got it. it did not look obvious in the first glance.

Reply:
I see. I moved the object to the top of the file and used uppercase for the keys. Hope this could help.